Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs

```diff
@@ -1489,5 +1490,551 @@ public void Including_an_interface_property_through_inheritance_should_work()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableProperty = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableProperty = 1 };
+
+            // Act
+            Action action =
+                () => subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_non_browsable_field_differs_excluding_non_browsable_members_should_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { NonBrowsableField = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { NonBrowsableField = 1 };
+
+            // Act & Assert
+            subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+        }
+
+        [Fact]
+        public void When_non_browsable_property_differs_excluding_non_browsable_members_should_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { NonBrowsableProperty = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { NonBrowsableProperty = 1 };
+
+            // Act & Assert
+            subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+        }
+
+        [Fact]
+        public void When_property_is_non_browsable_only_in_subject_excluding_non_browsable_members_should_not_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWhereMemberThatCouldBeNonBrowsableIsNonBrowsable() { PropertyThatMightBeNonBrowsable = 0 };
+            var expectation = new ClassWhereMemberThatCouldBeNonBrowsableIsBrowsable() { PropertyThatMightBeNonBrowsable = 1 };
+
+            // Act
+            Action action =
+                () => subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+
+            // Assert
+            action.Should().Throw<XunitException>().WithMessage("Expected property subject.PropertyThatMightBeNonBrowsable to be 1, but found 0.*");
+        }
+
+        [Fact]
+        public void When_property_is_non_browsable_only_in_subject_ignoring_non_browsable_members_on_subject_should_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWhereMemberThatCouldBeNonBrowsableIsNonBrowsable() { PropertyThatMightBeNonBrowsable = 0 };
+            var expectation = new ClassWhereMemberThatCouldBeNonBrowsableIsBrowsable() { PropertyThatMightBeNonBrowsable = 1 };
+
+            // Act & Assert
+            subject.Should().BeEquivalentTo(expectation, config => config.IgnoringNonBrowsableMembersOnSubject());
```

Comment:
🤔 This should throw right? Because the expectation has a property `PropertyThatMightBeNonBrowsable` that it will try to find on the subject.

Reply:
I thought members on the expectation that were missing on the subject were simply ignored.

Reply:
Or is it the other way around?

Reply:
Hmm, my use case requires a function like this which does _not_ throw. I don't want to use `.ExcludingMissingMembers` (which, I just noticed, is actually affecting how the _subject_ is processed despite having the word `Excluding` in its name 🤔), because then members that _aren't_ non-browsable that are missing won't cause test failures.

Reply:
I'm looking at the code in `StructuralEqualityEquivalencyStep` and I don't see why it would throw:
```
                IMember[] selectedMembers = GetMembersFromExpectation(context.CurrentNode, comparands, context.Options).ToArray();
                if (context.CurrentNode.IsRoot && !selectedMembers.Any())
                {
                    throw new InvalidOperationException(
                        "No members were found for comparison. " +
                        "Please specify some members to include in the comparison or choose a more meaningful assertion.");
                }

                foreach (IMember selectedMember in selectedMembers)
                {
                    AssertMemberEquality(comparands, context, nestedValidator, selectedMember, context.Options);
                }
```
So, it'll find all the members on the expectation, and then for each one it calls `AssertMemberEquality`:
```
        private static void AssertMemberEquality(Comparands comparands, IEquivalencyValidationContext context,
            IEquivalencyValidator parent, IMember selectedMember, IEquivalencyAssertionOptions options)
        {
            IMember matchingMember = FindMatchFor(selectedMember, context.CurrentNode, comparands.Subject, options);
            if (matchingMember is not null)
            {
                ...
            }
        }
```
If `FindMatchFor` returns `null` then it simply does nothing. Here's `FindMatchFor` before this PR:
```
        private static IMember FindMatchFor(IMember selectedMember, INode currentNode, object subject,
            IEquivalencyAssertionOptions config)
        {
            IEnumerable<IMember> query =
                from rule in config.MatchingRules
                let match = rule.Match(selectedMember, subject, currentNode, config)
                where match is not null
                select match;

            return query.FirstOrDefault();
        }
```
So, if the subject is missing a member found on the expectation, `FindMatchFor` returns `null` and `AssertMemberEquality` simply does nothing.

This PR extends `FindMatchFor` so that it won't find non-browsable members if `IgnoreNonBrowsableOnSubject` is set in `config`.

Am I missing something??

Reply:
Sorry, I'm lost now. It's probably all the double negatives in that sentence 🤔.

With respect to `ExcludingMissingMembers`, yes, it should have been called something like `IgnoringMissingMembers`. I think that in previous major versions the comparison was driven through the subject instead of the expectation.

Reply:
> I'm looking at the code in StructuralEqualityEquivalencyStep and I don't see why it would throw

Because `MustMatchByNameRule` will throw.

Reply:
Hmm, but in practice, it doesn't throw. The tests as written pass. 🤔

Reply:
Ohh, it's because of where it's excluding the non-browsable members. It does it _after_ the matching rule runs. So, `MustMatchByNameRule` doesn't throw because it _does_ find a matching member, but then the _call site_ throws it out.

Reply:
I've devised a fix for this. :-) It'll be in the next commit.

Reply:
And the test still works, it just has to include `.ExcludingMissingMembers()` in the config as well.